bundle/Routing/UrlAliasRouter.php

```diff
@@ -25,4 +68,181 @@ public function matchRequest(Request $request): array
+            return $parameters['location'];
+        }
+
+        if (isset($parameters['locationId'])) {
+            return $this->locationService->loadLocation($parameters['locationId']);
+        }
+
+        if (isset($parameters['contentId'])) {
+            $contentInfo = $this->contentService->loadContentInfo($parameters['contentId']);
+
+            if ($contentInfo->mainLocationId === null) {
+                throw new LogicException(
+                    'Cannot generate an UrlAlias route for content without the main Location'
+                );
+            }
+
+            return $this->locationService->loadLocation($contentInfo->mainLocationId);
+        }
+
+        throw new InvalidArgumentException(
+            'When generating an UrlAlias route, either "location", "locationId" or "contentId" parameter must be provided'
+        );
+    }
+
+    private function getFrontendSiteaccessNameForLocation(APILocation $location): string
+    {
+        $nameSet = $this->getFrontendSiteaccessNameSet($location);
+
+        if (empty($nameSet) || array_key_exists($this->currentSiteaccess->name, $nameSet)) {
+            return $this->currentSiteaccess->name;
+        }
+
+        return array_key_first($nameSet);
+    }
+
+    private function getFrontendSiteaccessNameSet(APILocation $location): array
+    {
+        if (array_key_exists($location->id, $this->locationIdFrontendSiteaccessNameSetMapCache)) {
+            return $this->locationIdFrontendSiteaccessNameSetMapCache[$location->id];
+        }
+
+        $ancestorLocationIds = array_map('\intval', $location->path);
+        $map = $this->getFrontendSiteaccessNameRootLocationIdMap();
+        $nameSet = [];
+
+        foreach ($map as $siteaccessName => $rootLocationId) {
+            if (in_array($rootLocationId, $ancestorLocationIds, true)) {
+                $nameSet[$siteaccessName] = true;
+            }
+        }
+
+        return $this->locationIdFrontendSiteaccessNameSetMapCache[$location->id] = $nameSet;
+    }
+
+    private function getFrontendSiteaccessNameRootLocationIdMap(): array
+    {
+        if ($this->frontendSiteaccessNameRootLocationIdMap !== null) {
+            return $this->frontendSiteaccessNameRootLocationIdMap;
+        }
+
+        $this->frontendSiteaccessNameRootLocationIdMap = [];
+
+        foreach ($this->siteaccessNames as $siteaccessName) {
+            if ($this->isAdminSiteaccess($siteaccessName)) {
+                continue;
+            }
+
+            $rootLocationId = $this->configResolver->getParameter(
+                'content.tree_root.location_id',
+                null,
+                $siteaccessName
+            );
+
+            $this->frontendSiteaccessNameRootLocationIdMap[$siteaccessName] = $rootLocationId;
+        }
+
+        return $this->frontendSiteaccessNameRootLocationIdMap;
+    }
+
+    private function isAdminSiteaccess(string $siteaccessName): bool
```

Comment:
I would not lock us in like this. I'd prefer to see this with a different naming and have configurable "deny list" for the siteaccess groups.

Reply:
Resolved through refactoring: https://github.com/netgen/ezplatform-site-api/pull/185/commits/4c350ba41938a79e096abd5fa9d592b35c4c571a